src/Security/LdapUserProvider.php

```diff
@@ -200,7 +200,7 @@ public function refreshUser(UserInterface $user)
      */
     public function supportsClass($class)
     {
-        return $class instanceof LdapUser;
+        return LdapUser::class === $class;
```

Comment:
What happens here if someone extends LdapUser and configures their own user class?

Reply:
That does not work, right. We then need to use `class_parents` here and check the inheritance here. I adjusted the method now because once we want to update to symfony 5, we are dealing with strict types in the interface: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Security/Core/User/UserProviderInterface.php#L68

So, I don't think going back to using an object here would be future proof. `class_parents` seems the only way to solve this.

Reply:
This is now solved as people are now allowed to overwrite the userprovider.